oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoVersionGCSupport.java

```diff
@@ -132,6 +133,49 @@ public CloseableIterable<NodeDocument> getPossiblyDeletedDocs(final long fromMod
                 input -> store.convertFromDBObject(NODES, input)));
     }
 
+    /**
+     * Calculate the bson representing including only the provided
+     * include path prefixes and/or excluding the provided
+     * exclude path prefixes - if any are provided - AND the provided
+     * query
+     * @param includes set of path prefixes which should only be considered
+     * @param excludes set of path prefixes which should be excluded.
+     * if these overlap with includes, then exclude has precedence.
+     * @param query the query with which to do an AND
+     * @return the combined bson with include/exclude path prefixes
+     * AND the provided query
+     */
+    private Bson withIncludeExcludes(Set<String> includes, Set<String> excludes, Bson query) {
+        Bson inclExcl = null;
+        if (includes != null && !includes.isEmpty()) {
+            final List<Bson> ors = new ArrayList<>(includes.size());
+            for (String incl : includes) {
+                ors.add(Filters.regex(ID, ":" + incl));
+            }
+            inclExcl = or(ors);
+        }
+        if (excludes != null && !excludes.isEmpty()) {
+            final List<Bson> ands = new ArrayList<>(excludes.size());
+            for (String excl : excludes) {
+                ands.add(not(Filters.regex(ID, ":" + excl)));
```

Comment:
Is this the part where we match oak paths to the identifiers used in a DocumentStore ("n:...")? In that case a comment would be good.

Also, do we need to consider the special identifiers for long paths? If we can't, excluding these documents in a layer above might work; but sure what that would mean for "includePaths" though.

Reply:
Yes, it is for the id syntax. Good point about long paths - the includes/excludes wouldn't have taken those into account as per current state of the PR. I'll look into that. (Worst case could be a comment saying that include/excludes aren't supported for long paths, but that would be unfortunate..)